docs/core/extensions/localization.md

```diff
@@ -0,0 +1,233 @@
+
+:::code source="snippets/localization/example/MessageService.cs" highlight="8,10-11,16":::
+
+In the preceding C# code:
+
+- A `IStringLocalizer<MessageService> _localizer` field is declared.
+- The constructor takes a `IStringLocalizer<MessageService>` parameter and assigns it to the `_localizer` field.
+- The `GetGreetingMessage` method invokes the <xref:Microsoft.Extensions.Localization.IStringLocalizer.Item(System.String)?displayProperty=nameWithType> passing `"GreetingMessage"` as an argument.
+
+The `IStringLocalizer` also supports parameterized string resources, consider the following `ParameterizedMessageService`:
+
+:::code source="snippets/localization/example/ParameterizedMessageService.cs" highlight="9,11-12,17":::
+
+In the preceding C# code:
+
+- A `IStringLocalizer _localizer` field is declared.
+- The constructor takes a `IStringLocalizerFactory` parameter, which is used to create a `IStringLocalizer` from the `ParameterizedMessageService` type, and assigns it to the `_localizer` field.
+- The `GetFormattedMessage` method invokes the <xref:Microsoft.Extensions.Localization.IStringLocalizer.Item(System.String,System.Object[])?displayProperty=nameWithType> passing `"DinnerPriceFormat"`, a `dateTime`, and the `dinnerPrice` as arguments.
+
+> [!IMPORTANT]
+> The `IStringLocalizerFactory` isn't required. Instead, it is preferred for consuming services to require the <xref:Microsoft.Extensions.Localization.IStringLocalizer%601>.
+
+Both <xref:Microsoft.Extensions.Localization.IStringLocalizer.Item%2A?displayProperty=nameWithType> indexers return a <xref:Microsoft.Extensions.Localization.LocalizedString>, which have [implicit conversions](xref:Microsoft.Extensions.Localization.LocalizedString.op_Implicit(Microsoft.Extensions.Localization.LocalizedString)~System.String) to `string?`.
+
+## Put it all together
+
+To exemplify an app using both message services, with localization and resource files consider the following *Program.cs* file:
+
+:::code source="snippets/localization/example/Program.cs":::
+
+In the preceding C# code:
+
+- The <xref:Microsoft.Extensions.Localization.RootNamespaceAttribute> sets `"Localization.Example"` as the root namespace.
+- The <xref:System.Console.OutputEncoding?displayProperty=nameWithType> is assigned to <xref:System.Text.Encoding.Unicode?displayProperty=nameWithType>.
+- When a single argument is passed to `args`, the <xref:System.Globalization.CultureInfo.CurrentCulture?displayProperty=nameWithType> and <xref:System.Globalization.CultureInfo.CurrentUICulture?displayProperty=nameWithType> are assigned the result of <xref:System.Globalization.CultureInfo.GetCultureInfo(System.String)?displayProperty=nameWithType> given the `arg[0]`.
+- The host is created with [defaults](generic-host.md#default-builder-settings), localization services, `MessageService`, and `ParameterizedMessageService` are all added to the `IServiceCollection` for DI.
```

Comment:
This sentence is grammatically incorrect. I'm not sure the intended meaning so not supplying a suggestion here.